std/machines/arith16.asm

```diff
@@ -10,45 +10,43 @@ use std::convert::expr;
 use std::prover::eval;
 use std::prelude::Query;
 use std::machines::range::Byte;
+use std::machines::range::Byte2;
 
-// Arithmetic machine, ported mainly from Polygon: https://github.com/0xPolygonHermez/zkevm-proverjs/blob/main/pil/arith.pil
-// This machine supports eq0, which is the affine equation. Currently we only expose operations for mul and div.
-machine Arith16(byte: Byte) with
+// Arithmetic machine, inspired by Polygon's 256-Bit Arith machine: https://github.com/0xPolygonHermez/zkevm-proverjs/blob/main/pil/arith.pil
+machine Arith16(byte: Byte, byte2: Byte2) with
     latch: CLK8_7,
-    operation_id: operation_id,
+    operation_id: is_division,
     // Allow this machine to be connected via a permutation
     call_selectors: sel,
 {
-    col witness operation_id;
-
-    // operation_id has to be either mul or div.
-    force_bool(operation_id);
+    col witness is_division;
 
     // Computes x1 * y1 + x2, where all inputs / outputs are 32-bit words (represented as 16-bit limbs in big-endian order).
-    // More precisely, affine_256(x1, y1, x2) = (y2, y3), where x1 * y1 + x2 = 2**16 * y2 + y3
-
-    // x1 * y1 = y2 * 2**16 + y3
-    operation mul<0> x1c[1], x1c[0], y1c[1], y1c[0] -> y2c[1], y2c[0], y3c[1], y3c[0];
-
-    // Constrain that x2 = 0 when operation is mul.
-    array::new(4, |i| (1 - operation_id) * x2[i] = 0);
+    // More precisely, affine(x1, y1, x2) = (y2, y3), where x1 * y1 + x2 = 2**16 * y2 + y3
+    operation mul<0> x1c[1], x1c[0], x2c[1], x2c[0], y1c[1], y1c[0] -> y2c[1], y2c[0], y3c[1], y3c[0];
     
     // y3 / x1 = y1 (remainder x2)
     // WARNING: it's not constrained that remainder is less than the divisor.
-    // This is done in the main machine, e.g. our RISCV BabyBear machine, that uses this operation.
+    // WARNING: For division by zero, the quotient is unconstrained.
+    // Both need to handled by any machine calling into this one.
```

Comment:
```suggestion
    // Both need to be handled by any machine calling into this one.
```

Reply:
I think these should be moved to the documentation of the operation

Reply:
I fixed the typo, but where do you want to move it? Isn't this the documentation of the operation?